src/web-components/KSLAddButtonElement.ts

```diff
@@ -228,10 +229,13 @@ export class KSLAddButtonElement extends KSLPositionedElement {
 
       const { elementType, permissions } = response;
 
-      if (!permissions || !hasAddButtonPermissions(permissions)) {
+      if (
+        !permissions ||
+        permissions.get(AddButtonPermission.ViewParent) === AddButtonPermissionCheckResult.PermissionMissing
```

Comment:
Why are you checking only 1 of all available permissions here? What about other permissions (edit, create new)? And what about other CheckResults (Unknown, NotTranslated)?

Reply:
**Checking only ViewParent permission**:

I was somewhat reluctant to implement the check in this simple way, but from what I understood in the _d.iii._ [here](https://kentico.atlassian.net/wiki/spaces/KPM/pages/2697068925/KCL-6719+Spike+notes+-+revised), we are currently not interested in any other permissions regarding the plus button itself than the "ViewParent" one, as one of the following will happen: 

- The user is missing ViewParent permission, plus button is disabled
- The user has ViewParent permission, plus button is enabled, toolbar will open on click
  - No matter the remaining permissions, all buttons in the toolbar will be enabled

There was a quick discussion at one of our dailies about whether we want to propagate the permission status into the toolbar buttons, but IIRC, in the end we've agreed that we don't want to do that in this task, and the [_d.iii._](https://kentico.atlassian.net/wiki/spaces/KPM/pages/2697068925/KCL-6719+Spike+notes+-+revised) point was updated accordingly, crossing out the "will be disabled" part. 

**Other CheckResults**: 

Oops, my mistake here, I think I intended to use `!== AddButtonPermissionCheckResult.Ok` here. Do we want the SDK to handle other non-ok results in a different ways (display different tooltip messages), or do we only care about disabling the plus button atm?

Reply:
I think that `Unknown` permission check result could be handled in the same way as `PermissionMissing`. 
When it comes to `NotTranslated`, I have kept that one in the model for the sake of completeness but realistically I don't see how a not translated item would even be rendered in a preview website.